src/libraries/Common/src/System/IO/MemoryMappedFileInternal.cs

```diff
@@ -0,0 +1,167 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
This file also contains code that is shared between the new methods and `MemoryMappedFile`. The code is also used in Unix, which is why the class had to be split as a partial.